R/sample_data.R

```diff
@@ -1,3 +1,65 @@
+#' Helper function to read the file and return a dataframe.
+#'
+#' Checks if file is type .csv or excel. If not, return an error.
+#'
+#' @param file the name of the file, including the path and filetype extension
+#' @param sheet_name if passing an excel file, the name of the sheet to analyze (default = NULL)
+#'
+#' @return dataframe
+#' @export
+load_file <- function(file, sheet_name = NULL) {
+  out <- tryCatch({
+    if (tools::file_ext(file) == "csv") {
+      readr::read_csv(file)
+    }
+    else {
+      readxl::read_xlsx(file, sheet = sheet_name)
+    }
+  },
+  error = function(cond) {
+    message("Please use a valid csv or excel file.")
+    message(cond)
+    return(NA)
+  })
+  return(out)
+}
+
+#' Helper function used to create summary data table showing column names in rows,
+#' an example record, column data types and summary statistics for different data types
+#'
+#' @param df the dataframe object to analyze
+#'
+#' @return dataframe
+#' @export
+summarize_data <- function(df) {
+  # create data frame with column names and first record
+  result <- as.data.frame(t(dplyr::slice(df, 1)))
+  result['columns'] = colnames(df)
+  result <- dplyr::select(dplyr::mutate(result, sample_record = V1), columns, sample_record)
```

Comment:
The generated .csv file doesn't show the `colname` and `sample_record` at my end

Reply:
Updated tests to make this change